:memo: [doc] Added README (#2)
perun-network/perun-eth-mobile: README.md

> ## Perun mobile bindings

Reviewer: h1

> This project provides Android bindings for *go-perun* called *prnm*.  

Author: Add link.

> Right now it only focusses on two-party-payment channels.  

Reviewer: ```suggestion
Right now it only provides on two-party-payment channels.  
```
(and "focuses" with one "s")

## Security Disclaimer
The authors take no responsibility for any loss of digital assets or other damage caused by the use of this software.  
**Do not use this software with real funds**.

### Getting Started
```sh
# Get perun-eth-mobile
go get -d https://github.com/perun-network/perun-eth-mobile
# Install gomobile, see https://godoc.org/golang.org/x/mobile/cmd/gomobile
go get golang.org/x/mobile/cmd/gomobile
gomobile init
# Generate the bindings
gomobile bind -o prnm.aar -target=android github.com/perun-network/perun-eth-mobile/
```

`prnm.aar` can then be included with Android studio.  
The easiest way of getting started with an app is to try out the *go-mobile* [example app](https://github.com/golang/go/wiki/Mobile#sdk-applications-and-generating-bindings) and instead of using the `hello.aar` replace it with `prnm.aar` that you generated with the command above.

### In Java

> Go to the `MainActivity.java` of your app and import `prnm.*`. Just to see whether or not the compilation and linking is working, you can try to call `Prnm.contextBackground()` which should return not nil.  

Reviewer: ```suggestion
Go to the `MainActivity.java` of your app and import `prnm.*`. Just to see whether or not the compilation and linking is working, you can try to call `Prnm.contextBackground()` which should return a non-null object.
```
Like this?

A sample setup for a two-party-payment channel could look like this:  
```java
// You can add a sleep here to ensure that the Android studio logger is attached.
// Thread.sleep(2000);
// I hope that this is the correct path for persistent files.
String appDir = getApplicationContext().getFilesDir().getAbsolutePath();
String ksPath = appDir +"/keystore";
String dbPath = appDir +"/database";

> // 10.0.2.2 is the IP of the host PC when using Android Simulator.

Reviewer: ```suggestion
// 10.0.2.2 is the IP of the host PC when using Android Simulator and the host is running a ganache-cli.
```

> // 8545 is the standart port for ganache-cli.

Reviewer: ```suggestion
// 8545 is the standart port of ganache-cli.
```

String ethUrl = "ws://10.0.2.2:8545";
// We are alice in this example and this is our on-chain address holding the ETH.

> Address alice = Prnm.newAddressFromHex("0x05e71027e7d3bd6261de7634cf50F0e2142067C4");

Reviewer: What about importing `Prnm` classes like `Address` and `Config`? Could we add a sample import statement at the top?

Author: That all works by `import prnm.*;` as i wrote above.

// Listen on 127.0.0.1:1337 for channel Proposals.

> Config cfg = Prnm.newConfig("Alice ", ksPath, "password", alice, dbPath, ethUrl, "127.0.0.1", 1337);

Reviewer: Can we create the `Config` class directly?

Author: The Java constructor syntax seems to translate well to global functions, also using it for `Address` and `Client` now.

// This is currently missing the ProposalHandler.
Client client = Prnm.newClient(cfg);
// PerunId (currently an Address) of the peer that we want to open a channel with.
Address bob = Prnm.newAddressFromHex("0xA298Fc05bccff341f340a11FffA30567a00e651f");
// Tell the client what `bob`s ip and port are.

> client.addPeer(bob, "10.0.2.2", 1338);

Reviewer: Since alice and bob are on different hosts in general, we should maybe use the same port 1337 to avoid confusion? I also think that 1337 is quite overused for examples, to avoid clashes, we should invent some Perun port range, like 5750, 5751, ... which is also not in the list of ports at https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers

Author: Yea that was a residue from the node, we can use 5750 i guess.

// Create the initial balances of the channel, me starting with 2000 and bob with 1000.
Allocation initBals = Prnm.newAllocation(Prnm.newBigIntFromInt64(2000),Prnm.newBigIntFromInt64(1000));
// The channel proposal does not time out (Don't do this in production).

> Context ctx = Prnm.background();

Reviewer: please rename (in other PR) to `ContextBackground()`

Reviewer: The user will probably greatly appreciate a `Prnm.ContextWithDuration` or `Timeout`...
Maybe at least use a `ContextWithCancel` here and write in the comment that `ctx.cancel()` can be called to cancel the ongoing request.

Author: But then it also must be cancelled, we should add that as comment.

Reviewer: Yes, add a comment that `ctx.Cancel()` can be called to cancel the ongoing operation. Do you also refer to the ressource leak? True, it should also be mentioned in the comment that `ctx.Cancel()` also needs to be called in a surrounding `finally{}` block to release the resources of this cancel context.

I think it is important to mention this here, because if we just use ContextBackground the whole point is missing what those weird contexts are for.

// Propose a channel to `bob` with `initBals` and 60s challenge duration.
PaymentChannel channel = client.proposeChannel(ctx, bob, initBals, 60);
```  
This code must be wrapped in a `try` as the compiler will let you know.  

> You can always `Ctrl+Click` on a *prnm* function to use the Android Studio decompiler wich is really helpfull to see all the available Java functions.

Reviewer: ```suggestion
You can always `Ctrl+Click` on a *prnm* function to use the Android Studio decompiler, wich is really helpful to see all available Java functions.
```

> -> This is missing the `ChannelProposal` and `UpdateResponder` handlers right now.  

Reviewer: ```suggestion
The implementation of incoming channel proposals and channel updates will be added shortly.
```
